markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -96,77 +99,120 @@ library PerpsMarket {
-                .min(maxAllowedLiquidationInWindow, requestedLiquidationAmount)
-                .to128();
-            self.lastUtilizedLiquidationCapacity = liquidatableAmount;
-        } else {
-            liquidatableAmount = MathUtil
-                .min(
-                    maxAllowedLiquidationInWindow - self.lastUtilizedLiquidationCapacity,
-                    requestedLiquidationAmount
-                )
-                .to128();
-            self.lastUtilizedLiquidationCapacity += liquidatableAmount;
+        (
+            uint liquidationCapacity,
+            uint maxLiquidationInWindow,
+            uint128 latestLiquidationTimestamp
+        ) = currentLiquidationCapacity(self, marketConfig);
+
+        // this would only occur if there was a misconfiguration (like skew scale not being set)
+        // or the max liquidation window not being set etc.
+        // in this case, return the entire requested liquidation amount
+        if (maxLiquidationInWindow == 0) {
+            return requestedLiquidationAmount;
         }
 
-        // liquidatable amount is 0 only if there's no more capacity in the window
-        // but if maxLiquidationPd is set, and the current market p/d is less than that,
-        // then allow for an extra block of liquidation
         uint maxLiquidationPd = marketConfig.maxLiquidationPd;
-        if (
-            liquidatableAmount == 0 &&
+        // if liquidation capacity exists, update accordingly
+        if (liquidationCapacity != 0) {
+            liquidatableAmount = MathUtil.min128(
+                liquidationCapacity.to128(),
+                requestedLiquidationAmount
+            );
+        } else if (
             maxLiquidationPd != 0 &&
             // only allow this if the last update was not in the current block
-            self.lastTimeLiquidationCapacityUpdated != block.timestamp.to128()
+            latestLiquidationTimestamp != block.timestamp
         ) {
+            /**
+                if capacity is at 0, but the market is under configured liquidation p/d,
+                another block of liquidation becomes allowable.
+             */
             uint256 currentPd = MathUtil.abs(self.skew).divDecimal(marketConfig.skewScale);
             if (currentPd < maxLiquidationPd) {
-                liquidatableAmount = (
-                    requestedLiquidationAmount > maxAllowedLiquidationInWindow
-                        ? maxAllowedLiquidationInWindow
-                        : requestedLiquidationAmount
-                ).to128();
-                // track how much was utilized in this block to ensure any subsequent
-                // liquidations don't combine for larger than what's allotted in the window
-                self.lastUtilizedLiquidationCapacity = liquidatableAmount;
+                liquidatableAmount = MathUtil.min128(
+                    maxLiquidationInWindow.to128(),
+                    requestedLiquidationAmount
```

Comment:
If the value will (usually) depend on maker, taker and skewScale. We can keep the setting as kind of read-only and update the value each time any of maker, taker or skewScale is updated. But TBH, I prefer it to be an independent configuration parameter that we can tweak if needed later. i.e. to be more restrictive or less restrictive on a specific market

Reply:
Yeah i am ok with having it configurable on that variable @leomassazza , no issue, i'm just saying, that we can only liquidate up to reaching a certain point, 
Ok so to figure out the limit:
abs(currentPD - maxPD)*skewScale
So as an example: 
- currentPD is 10 bp
- maxPD is 30 bp
- skewScale is 1m ETH
Liquidations allowed is up to the point where we reach maxPD, which is (20-10)*1m = 1000 ETH

Reply:
`maxLiquidationAmount` is not a configured value, it's derived.  it can stay that way, and then in the logic where currentPd < configuredPd, we _could_ limit it until we hit the configured pd and only liquidate that amount...or we just let them liquidate the max amount for the window.  i'll sketch out the change if we were to go with teh former.

Reply:
@kaleb-keny chatted with afif about this.  as long as you're under p/d, we are ok allowing upto max amount per window even if this max puts the p/d above the configured threshold.  on subsequent liquidation calls, no more will be allowed.